tooling/cli/src/mobile/mod.rs

```diff
@@ -362,6 +365,31 @@ fn ensure_init(
   Ok(())
 }
 
+#[cfg(unix)]
+fn ensure_gradlew(project_dir: &std::path::Path) -> Result<()> {
+  use std::os::unix::fs::PermissionsExt;
+
+  let gradlew_path = project_dir.join("gradlew");
+  if let Ok(metadata) = gradlew_path.metadata() {
+    let mut permissions = metadata.permissions();
+    let is_executable = permissions.mode() & 0o111 != 0;
+    if !is_executable {
+      permissions.set_mode(0o755);
```

Comment:
should we preserve previous permissions and only add `0o111` for execution bit? instead of always setting `0o755`